apis/apiextensions/v1alpha1/crds.go

```diff
@@ -0,0 +1,277 @@
+          made, if any. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency'
+        type: string
+      uid:
+        description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
+        type: string
+    required:
+    - apiVersion
+    - kind
+    - name
+  type: array
+writeConnectionSecretToRef:
+  description: WriteConnectionSecretsToNamespace specifies the namespace
+    in which the connection secret of the composite resource will be created.
+  properties:
+    name:
+      description: Name of the secret.
+      type: string
+  required:
+  - name
+  type: object
+`
+
+// InfraCompositeStatusProps is OpenAPIV3Schema for status fields that Crossplane uses for
+// Infrastructure kinds.
+const InfraCompositeStatusProps = `
+bindingPhase:
+  description: Phase represents the binding phase of a managed resource
+    or claim. Unbindable resources cannot be bound, typically because
+    they are currently unavailable, or still being created. Unbound resource
+    are available for binding, and Bound resources have successfully bound
+    to another resource.
+  enum:
+  - Unbindable
+  - Unbound
+  - Bound
+  - Released
+  type: string
+conditions:
+  description: Conditions of the resource.
+  items:
+    description: A Condition that may apply to a resource.
+    properties:
+      lastTransitionTime:
+        description: LastTransitionTime is the last time this condition
+          transitioned from one status to another.
+        format: date-time
+        type: string
+      message:
+        description: A Message containing details about this condition's
+          last transition from one status to another, if any.
+        type: string
+      reason:
+        description: A Reason for this condition's last transition from
+          one status to another.
+        type: string
+      status:
+        description: Status of this condition; is it currently True, False,
+          or Unknown?
+        type: string
+      type:
+        description: Type of this condition. At most one of each condition
+          type may apply to a resource at any point in time.
+        type: string
+    required:
+    - lastTransitionTime
+    - reason
+    - status
+    - type
+    type: object
+  type: array
+`
+
+const (
+	errConvertCRDTemplate = "cannot convert given crd spec template into actual crd spec"
+)
+
+// NOTE(muvaf): We use v1beta1.CustomResourceDefinition for backward compatibility
+// with clusters pre-1.16
+
+// TODO(muvaf): Every field on top level spec could be a DefinitionOption that is
+// reused, although it is known that only two different kinds will be generated.
+
+// BaseCRD returns a base template for generating a CRD.
+func BaseCRD(opts ...func(*v1beta1.CustomResourceDefinition) error) (*v1beta1.CustomResourceDefinition, error) {
```

Comment:
I did that originally but since it's in `apis` folder, controller-gen returns error saying the type is not compatible. I tried to find a way to skip that type but as far as I see, you can only skip [the whole package](https://book.kubebuilder.io/reference/markers/crd.html).